### PR TITLE
Implement H&S delete

### PR DESCRIPTION
**Pull request by sergii-iakovenko**
The changes:
- Implement H&S delete via V2 API.
- Fix switch rule removal in H&S reroute (the issue with remove commands for an one-switch flow).

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/2869)

### CODE REVIEW

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/delete/actions/CompleteFlowPathRemovalAction.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                if (path.isForward()) {
+                    // Try to remove as paired first
+                    try {
+                        PathId oppositePathId = flow.getOppositePathId(path.getPathId());
+                        if (oppositePathId != null) {
+                            Optional<FlowPath> oppositePath = flow.getPath(oppositePathId);
+                            if (oppositePath.isPresent()) {
+                                log.debug("Removing the flow paths {} / {}", path.getPathId(), oppositePathId);
+                                flowPathRepository.delete(path);
+                                flowPathRepository.delete(oppositePath.get());
```
Comment (niksv) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/delete/actions/CompleteFlowPathRemovalAction.java]: Do we need to `updateIslsForFlowPath` here?

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/delete/actions/RemoveRulesAction.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        if (protectedForward != null && protectedReverse != null) {
+            pathIds.remove(protectedForward.getPathId());
+            pathIds.remove(protectedReverse.getPathId());
+
+            stateMachine.addFlowResources(buildResources(flow, protectedForward, protectedReverse));
+
+            commands.addAll(commandBuilder.createRemoveNonIngressRules(
+                    stateMachine.getCommandContext(), flow, protectedForward, protectedReverse));
+            commands.addAll(commandBuilder.createRemoveIngressRules(
+                    stateMachine.getCommandContext(), flow, protectedForward, protectedReverse));
```
Comment (niksv) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/delete/actions/RemoveRulesAction.java]: AFAIK protected paths have on Ingress rules. Only transit and egress

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java [hunk 3]
```diff
@@ -308,6 +308,25 @@
...
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CompletableFuture<FlowResponseV2> deleteFlowV2(String flowId) {
+        logger.info("Delete flow request for flow: {}", flowId);
+
+        FlowDto flow = new FlowDto();
+        flow.setFlowId(flowId);
+        CommandMessage command = new CommandMessage(new FlowDeleteRequest(flow),
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java]: Agree. Fixed

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/delete/actions/CompleteFlowPathRemovalAction.java (hunk 1 above)
Comment (sergii-iakovenko) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/delete/actions/CompleteFlowPathRemovalAction.java]: Nice catch! Fixed, although it has been refactored in h-n-s-update.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/delete/actions/RemoveRulesAction.java (hunk 2 above)
Comment (sergii-iakovenko) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/delete/actions/RemoveRulesAction.java]: Right. Fixed

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/FlowDeleteRequest.java [hunk 4]
```diff
@@ -15,103 +15,32 @@
...
      */
     private static final long serialVersionUID = 1L;
 
-    /**
-     * The request payload.
-     */
-    @JsonProperty(PAYLOAD)
-    protected FlowDto payload;
+    @JsonProperty("flow_id")
+    protected String flowId;
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/FlowDeleteRequest.java]: Fixed

**Pull request merged**